config/index.ts

```diff
@@ -81,13 +81,38 @@ export function writeConfig(options: WriteConfigOptions = {}): void {
   }
 }
 
+export function getRootOrDeprecatedConfigPath(
+  useRootConfig = false
+): string | null {
+  if (useRootConfig || CLIConfiguration.isActive()) {
+    return getConfigFilePath();
+  }
+  return config_DEPRECATED.getConfigPath();
+}
+
 export function getConfigPath(path?: string): string | null {
   if (CLIConfiguration.isActive()) {
     return getConfigFilePath();
   }
   return config_DEPRECATED.getConfigPath(path);
 }
 
+export function bothConfigFilesExist(useRootConfig?: boolean): boolean {
+  if (configFileExists() && config_DEPRECATED.getConfigPath()) {
+    return true;
+  }
+
+  if (useRootConfig && config_DEPRECATED.getConfigPath()) {
+    return true;
+  }
+
+  if (!useRootConfig && configFileExists()) {
+    return true;
+  }
+
+  return false;
+}
+
```

Comment:
I think there may be a flaw in the logic here. Or at least, it's not actually checking if both config files exist, in which case, it might make sense to rename it.

The first check for both config files existing effectively isn't doing anything (other than short circuiting, but I don't think thats really worth it at the expense of the logic being harder to follow), since if both exist, one of the following conditions will always be true.

If the logic here is working as intended, I _think_ this could be simplified to
```
return useRootConfig ? Boolean(config_DEPRECATED.getConfigPath()) : configFileExists()
```

Reply:
Based on how this is used in `hs auth`, it looks like it'll return true whenever `configFileExists()` is true

Reply:
Thank you, Camden, this was really helpful and helped me get my head on straight.

Reply:
Addressed in https://github.com/HubSpot/hubspot-local-dev-lib/pull/172/commits/c14b1aa085ea0995951a0a73da3599756b91415b